internal/task_documentation/task_documentation.go

```
@@ -0,0 +1,146 @@
+
+	// YAML
+	if *yamlDir != "" {
+		if err = os.MkdirAll(*yamlDir, DirectoryPermissions); err != nil {
+			return
+		}
+		if err := findTektonTasks(*startPath, *taskDir, *yamlDir); err != nil {
+			return
+		}
+	}
+}
+
+func findTektonTasks(basePath, taskDir, dstDir string) error {
+	var err error
+	if basePath == "" {
+		basePath, err = getGitRoot()
+		if err != nil {
+			return fmt.Errorf("no base path provided and there was an error getting Git root: %v", err)
+		}
+	}
+
+	taskPath := filepath.Join(basePath, taskDir)
+	err = filepath.Walk(taskPath, func(currentPath string, info os.FileInfo, err error) error {
+		if err != nil {
+			return err
+		}
+		if !info.IsDir() && strings.HasSuffix(currentPath, ".yaml") {
+			isTektonTask, err := isTektonTask(currentPath)
+			if err != nil {
+				return err
+			}
+			if isTektonTask {
+				err = copyFile(currentPath, filepath.Join(basePath, dstDir, info.Name()))
+				if err != nil {
+					return err
+				}
+			}
+		}
+		return nil
+	})
+	return err
+}
+
+func getGitRoot() (string, error) {
+	cmd := exec.Command("git", "rev-parse", "--show-toplevel")
+	output, err := cmd.Output()
+	if err != nil {
+		return "", err
+	}
+	scanner := bufio.NewScanner(strings.NewReader(string(output)))
+	for scanner.Scan() {
+		return scanner.Text(), nil
+	}
+	return "", nil
+}
+
+func copyFile(src, dst string) error {
+	fmt.Printf("Copying %s to %s\n", src, dst)
+	in, err := os.Open(src)
+	if err != nil {
+		return err
+	}
+	defer in.Close()
+
+	out, err := os.Create(dst)
+	if err != nil {
+		return err
+	}
+	defer out.Close()
+
+	_, err = io.Copy(out, in)
+	if err != nil {
+		return err
+	}
+	return out.Close()
+}
+
+type TektonTask struct {
+	APIVersion string `yaml:"apiVersion"`
+	Kind       string `yaml:"kind"`
+}
+
+func isTektonTask(file string) (bool, error) {
+	task := TektonTask{}
+
+	data, err := os.ReadFile(file)
+	if err != nil {
+		return false, err
+	}
+
+	err = yaml.Unmarshal(data, &task)
+	if err != nil {
+		return false, err
```

If it's a yaml file, but not a kubernetes resource, do we want to return an error? Maybe swallowing the error here makes sense.

---

I'd prefer to leave it here since the caller can determine if something should be done with the error (maybe the file disappeared from underneath us?, etc)